src/modules/builtin/mv.rs

```diff
@@ -0,0 +1,61 @@
+use std::mem::swap;
+
+use heraclitus_compiler::prelude::*;
+use crate::modules::expression::expr::Expr;
+use crate::modules::condition::failed::Failed;
+use crate::translate::module::TranslateModule;
+use crate::docs::module::DocumentationModule;
+use crate::utils::{ParserMetadata, TranslateMetadata};
+use crate::modules::command::modifier::CommandModifier;
+
+#[derive(Debug, Clone)]
+pub struct Mv {
+    source: Expr,
+    destination: Expr,
+    modifier: CommandModifier,
+    failed: Failed,
+}
+
+impl SyntaxModule<ParserMetadata> for Mv {
+    syntax_name!("MoveFiles");
+
+    fn new() -> Self {
+        Mv {
+            source: Expr::new(),
+            destination: Expr::new(),
+            failed: Failed::new(),
+            modifier: CommandModifier::new().parse_expr(),
+        }
+    }
+
+    fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
+        syntax(meta, &mut self.modifier)?;
+        self.modifier.use_modifiers(meta, |_this, meta| {
+            token(meta, "mv")?;
+            syntax(meta, &mut self.source)?;
+            syntax(meta, &mut self.destination)?;
+            match syntax(meta, &mut self.failed) {
+                Ok(_) => Ok(()),
+                Err(err) => Err(err)
+            }
```

Comment:
One nitpick:
```suggestion
            syntax(meta, &mut self.failed)?
```
This is simpler, isn't it?

Reply:
Yeah you are right, updated the pr.